math/src/elliptic_curve/short_weierstrass/curves/bls12_381/field_extension.rs

```diff
@@ -65,14 +66,14 @@ impl IsField for Degree2ExtensionField {
 
     /// Returns the multiplicative inverse of `a`
     /// This uses the equality `(a0 + a1 * t) * (a0 - a1 * t) = a0.pow(2) - a1.pow(2) * Q::residue()`
-    fn inv(a: &Self::BaseType) -> Self::BaseType {
-        let inv_norm = (a[0].pow(2_u64) + a[1].pow(2_u64)).inv();
-        [&a[0] * &inv_norm, -&a[1] * inv_norm]
+    fn inv(a: &Self::BaseType) -> Result<Self::BaseType, FieldError> {
+        let inv_norm = (a[0].pow(2_u64) + a[1].pow(2_u64)).inv().unwrap();
```

Comment:
Is `a[0].pow(2_u64) + a[1].pow(2_u64)` guaranteed to never be `0` in the field? If so, please add a `SAFETY:` comment explaining why. Otherwise, this should bubble up the error.

Reply:
Sounds good